mono/mini/interp/transform.c

```diff
@@ -1008,6 +1009,8 @@ interp_handle_intrinsics (TransformData *td, MonoMethod *target_method, MonoMeth
 				if (!mini_magic_is_int_type (src) || !mini_magic_is_int_type (dst)) {
 					if (mini_magic_is_int_type (src))
 						store_value_as_local = TRUE;
+					else if (mono_class_is_magic_float (src_klass))
```

Comment:
Why not use mini_magic_is_float_type  ?

Reply:
it accepts `r4`, `r8` and `nfloat`. here I really only wanna do that if it's `nfloat`.